lib/persist-state.c

```diff
@@ -889,26 +890,43 @@ void
 persist_state_alloc_string(PersistState *self, const gchar *persist_name, const gchar *value, gssize len)
 {
   PersistEntryHandle handle;
-  SerializeArchive *sa;
+  SerializeArchive *serializer;
   GString *buf;
+  gsize buflen;
   gboolean success;
   gpointer block;
 
+  PersistEntryHandle lookup_handle;
+  gsize lookup_size;
+  guint8 lookup_result_version;
+  gpointer lookup_block;
```

Comment:
Feel free to declare variables where you assign values to them first. It might be a matter of preference/taste, I just wanted to let you know with this comment that these old-style declarations at the top of functions are not requirements.

Reply:
Thank you for letting me know! For now I'll leave it as is to avoid two different styles within a single function.